### ISSUE TITLE
Unify operations that can be resolved by higher level ops

### ISSUE THREAD

**Issue by dgomezTT** [tenstorrent/pytorch2.0_ttnn#992]
A good example of the linear operation for bert, or the softmax whose patterns are repeated all over the compiled code and can be unified under a ttnn op, like linear or attention_softmax_

New tickets will be created for future unifications

The goal of this is to improve performance
